### PR TITLE
Clear function

### LINKED ISSUE #43: Add clear all method
fantastic library , can you add a drop all/clear method. 

Say after n period of time, I want to clear out everything but not resize down .

### PR DESCRIPTION
**Pull request by semihbkgr**
fix #43

### CODE REVIEW

File: map.go [hunk 1]
```diff
@@ -350,6 +351,23 @@ func (m *Map[K, V]) Grow(newSize uintptr) {
 	}
 }
 
+// Clear the map by removing all entries in the map.
+// This operation resets the underlying metadata to its initial state.
+func (m *Map[K, V]) Clear() {
+	index := make([]*element[K, V], defaultSize)
```
Comment (alphadose) [inline map.go:357]:
```suggestion
	index := make([]*element[K, V], m.defaultSize)
```

File: e2e_test.go [hunk 2]
```diff
@@ -235,6 +235,31 @@ func TestIterator(t *testing.T) {
...
+	log := int(math.Log2(defaultSize))
+	expectedSize := uintptr(strconv.IntSize - log)
+	if m.metadata.Load().keyshifts != expectedSize {
+		t.Error("keyshift is not as expected after clear")
+	}
+	for i := 0; i < 100; i++ {
+		_, ok := m.Get(i)
+		if ok {
+			t.Error("the entries should not be existing in the map after clear")
+		}
```
Comment (alphadose) [inline e2e_test.go:259]:
```suggestion
		if _, ok := m.Get(i); ok {
			t.Error("the entries should not be existing in the map after clear")
		}
```

File: map.go [hunk 3]
```diff
@@ -350,6 +351,23 @@ func (m *Map[K, V]) Grow(newSize uintptr) {
...
+	header := (*reflect.SliceHeader)(unsafe.Pointer(&index))
+
```
Comment (alphadose) [inline map.go:359]: can you remove the blank line here ?

File: map.go [hunk 4]
```diff
@@ -350,6 +351,23 @@ func (m *Map[K, V]) Grow(newSize uintptr) {
...
+	newdata := &metadata[K, V]{
+		keyshifts: strconv.IntSize - log2(m.defaultSize),
+		data:      unsafe.Pointer(header.Data),
+		index:     index,
+	}
+
```
Comment (alphadose) [inline map.go:365]: can you remove the blank line here as well ?

**Comment by Tochemey** [comment]
@alphadose please any update on this?

**Review by alphadose — APPROVED** [review]
apologies, I had completely forgotten about this

**Comment by alphadose** [comment]
https://github.com/alphadose/haxmap/releases/tag/v1.4.0

**Pull request merged**